test/integration/ls.bats

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bats
+
+load helpers
+
+teardown() {
+  echo "$BATS_TEST_NAME
+----------
+$output
+----------
+
+" >> ${BATS_LOG}
+  machine rm -f testmachine
+}
+
+@test "ls: filter on driver" {
+  run machine create -d none --url tcp://127.0.0.1:2375 testmachine
+  run machine ls --filter driver=none
+  [ "$status" -eq 0  ]
+  [[ ${lines[1]} =~ "testmachine" ]]
+}
+
+@test "ls: filter on swarm" {
+  run machine create -d none --url tcp://127.0.0.1:2375 --swarm --swarm-master --swarm-discovery deadbeef testmachine
```

Comment:
oh, and I just realized that was missing a `token://`. I suppose it works because it's the `none` driver, but I'll fix it anyway just in case someone uses it as an example some day :wink: